src/main/java/edu/unc/lib/boxc/migration/cdm/validators/PermissionsValidator.java

```diff
@@ -0,0 +1,118 @@
+package edu.unc.lib.boxc.migration.cdm.validators;
+
+import edu.unc.lib.boxc.auth.api.UserRole;
+import edu.unc.lib.boxc.migration.cdm.exceptions.MigrationException;
+import edu.unc.lib.boxc.migration.cdm.model.MigrationProject;
+import edu.unc.lib.boxc.migration.cdm.model.PermissionsInfo;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PermissionsValidator {
+    private MigrationProject project;
+    private List<String> patronRoles;
+
+    /**
+     * Validate the permissions mappings for this project.
+     * @param force if true, incomplete, overlapping and duplicate mappings will be ignored
+     * @return A list of errors. An empty list will be returned for a valid mapping
+     */
+    public List<String> validateMappings(boolean force) {
```

Comment:
Is the validator going to be used before sip generation? I'm not sure it makes sense to provide a force flag for just the validate command. Also, i don't think the force flag is doing everything listed in the java doc, it only seems to be used for ignoring blank ids. Maybe it can be removed, or fully implemented if its needed?